tests/zquantum/core/cost_function_test.py

```diff
@@ -25,223 +26,243 @@
-    assert np.isclose(total.precision, np.sqrt((1 + 0.5 + 0.5 + 2 + 7) / 10))
-
-
-@pytest.fixture
-def ansatz_based_cost_function():
-    target_operator = QubitOperator("Z0")
-    ansatz = MockAnsatz(number_of_layers=1, problem_size=1)
-    backend = SymbolicSimulator()
-    estimation_method = estimate_expectation_values_by_averaging
-    estimation_preprocessors = [partial(allocate_shots_uniformly, number_of_shots=1)]
-    return AnsatzBasedCostFunction(
-        target_operator,
-        ansatz,
-        backend,
-        estimation_method=estimation_method,
-        estimation_preprocessors=estimation_preprocessors,
-    )
+        # and if only everything went right, this only call should be of the form
+        # noisy_ansatz.ansatz.get_executable_circuit(params+noise)
+        # Therefore, we extract the single argument and compare it to the
+        # expected one.
+        assert np.array_equal(
+            parametrized_circuit.bind.call_args[0][0], expected_symbols_map
+        )
 
+        # Note, normally, we would just do it in a single assert:
+        # noisy_ansatz.ansatz.get_executable_circuit.assert_called_once_with(params_noise)
+        # However, this does not work with numpy arrays, as it uses == operator
+        # to compare arguments, which does not produce boolean value for numpy arrays
 
-def test_ansatz_based_cost_function_returns_value_between_plus_and_minus_one(
-    ansatz_based_cost_function,
-):
-    params = np.array([1])
-    value = ansatz_based_cost_function(params)
-    assert -1 <= value <= 1
 
+class TestSumExpectationValues:
+    def test_sum_expectation_values(self):
+        expectation_values = ExpectationValues(np.array([5, -2, 1]))
+        total = sum_expectation_values(expectation_values)
+        assert np.isclose(total.value, 4)
+        assert total.precision is None
 
-@pytest.fixture
-def noisy_ansatz_cost_function_with_ansatz():
-    target_operator = QubitOperator("Z0")
-    ansatz = MockAnsatz(number_of_layers=2, problem_size=1)
-    backend = SymbolicSimulator()
-    estimation_method = mock.Mock(wraps=calculate_exact_expectation_values)
-    return (
-        AnsatzBasedCostFunction(
-            target_operator,
-            ansatz,
-            backend,
-            estimation_method=estimation_method,
-            parameter_precision=1e-4,
-            parameter_precision_seed=RNGSEED,
-        ),
-        ansatz,
+    def test_sum_expectation_values_with_covariances(self):
+        values = np.array([5, -2, 1])
+        correlations = [np.array([[1, 0.5], [0.5, 2]]), np.array([[7]])]
+        covariances = [correlations[0] / 10, correlations[1] / 10]
+        expectation_values = ExpectationValues(values, correlations, covariances)
+        total = sum_expectation_values(expectation_values)
+        assert np.isclose(total.value, 4)
+        assert np.isclose(total.precision, np.sqrt((1 + 0.5 + 0.5 + 2 + 7) / 10))
+
+
+TARGET_OPERATOR = QubitOperator("Z0")
```

Comment:
Why do we use global variables instead of pytest fixtures?

Reply:
I thought they are simpler and made more sense to use when the fixture/variable value is a simple 1 line statement (@alexjuda also holds this view) As a bonus, they do not require frustrated debugging of `Fixture called directly` errors.

Reply:
Ok!